### PR TITLE
Two amends to installation and usage docs: SELinux and SCCs

### LINKED ISSUE #686: Some steps to generate and apply SELinux policies are missing from current documentation
Some steps to generate and apply SELinux policies are missing from current documentation

Added a step-by-step demo with SELinux to
https://github.com/marcredhat/ocplibvirt/blob/main/README.md

It covers everything from
setting up OpenShift on libvirt to
deploying the Security Profiles Operator and a sample app
creating an SELinux RecordingProfile
applying the generated SELinux policy
checking on the OpenShift nodes that the sample app's container run with the correct SELinux type
etc

Hope this helps,
Marc

### PR DESCRIPTION
**Pull request by jhrozek**
- doc: Update SPO installation_usage with some SELinux specific documentation
- doc: Add OCP specific docs

#### What type of PR is this?
/kind documentation

#### What this PR does / why we need it:
There's two patches in this PR. One adds at least some documentation about using
and recording SELinux policies and the other adds OpenShift-specific chapters
that help users avoid some common issues related to OpenShift's use of SCCs.

#### Which issue(s) this PR fixes:
Fixes #686

#### Does this PR have test?
N/A

#### Special notes for your reviewer:
N/A

#### Does this PR introduce a user-facing change?

```release-note
NONE
```

### CODE REVIEW

File: installation-usage.md [hunk 1]
```diff
@@ -10,14 +10,17 @@
...
-  - [Apply profile to pod](#apply-profile-to-pod)
+- [Create a seccomp profile](#create-a-seccomp-profile)
+  - [Apply a seccomp profile to a pod](#apply-a-seccomp-profile-to-a-pod)
   - [Base syscalls for a container runtime](#base-syscalls-for-a-container-runtime)
   - [Bind workloads to profiles with ProfileBindings](#bind-workloads-to-profiles-with-profilebindings)
   - [Record profiles from workloads with <code>ProfileRecordings</code>](#record-profiles-from-workloads-with-profilerecordings)
     - [Hook based recording](#hook-based-recording)
     - [Log enricher based recording](#log-enricher-based-recording)
     - [eBPF based recording](#ebpf-based-recording)
+- [Create a SELinux profile](#create-a-selinux-profile)
```
Comment (saschagrunert) [inline installation-usage.md:21]:
```suggestion
- [Create a SELinux Profile](#create-a-selinux-profile)
```

File: installation-usage.md [hunk 2]
```diff
@@ -1142,6 +1255,134 @@ bpf-recorder. To use a custom path for `/proc`, just patch the spod accordingly:
...
+- downwardAPI
+- emptyDir
+- persistentVolumeClaim
+- projected
+- secret
+```
+
+Please note that a common mistake when creating custom SCCs is to bind them to a wide range of users or SAs
+through the `group` attribute, e.g. the `system:authenticated` group. Make sure your SCC is only usable by
+the serviceAccount it is supposed to be used by. Please refer to the OCP documentation (https://docs.openshift.com/container-platform/4.9/authentication/managing-security-context-constraints.html)
```
Comment (saschagrunert) [inline installation-usage.md:1337]:
```suggestion
the serviceAccount it is supposed to be used by. Please refer to the [OCP documentation](https://docs.openshift.com/container-platform/4.9/authentication/managing-security-context-constraints.html)
```

File: installation-usage.md [hunk 3]
```diff
@@ -1142,6 +1255,134 @@ bpf-recorder. To use a custom path for `/proc`, just patch the spod accordingly:
...
+or this Red Hat blog post (https://cloud.redhat.com/blog/managing-sccs-in-openshift) for more information
```
Comment (saschagrunert) [inline installation-usage.md:1338]:
```suggestion
or [this Red Hat blog post](https://cloud.redhat.com/blog/managing-sccs-in-openshift) for more information
```

**Comment by jhrozek** [comment]
thanks, for the review, I applied the suggestions using the GH UI, which I never did before so I hope I didn't break something :-)

**Review by saschagrunert — APPROVED** [review]

**Review by JAORMX — APPROVED** [review]

**Comment by jhrozek** [comment]
Let's just merge this, it's only amending the docs, no reason to waste watts re-running CI jobs..

**Pull request merged**